### PR TITLE
feat(supportplatform-service): create evaluation method backend crud

### LINKED ISSUE #133: [SupportPlatform-Service] Implement create, update, read, delete, unit and integration tests of evaluative methods

### LINKED ISSUE #167: [SupportPlatform-Service] [JobScheduler-Service] Fix "Identifier name is too long" to PullRequestHasAssigneContributor index and IssueHasAssigneContributor index

### LINKED ISSUE #168: [SupportPlatform-Service] [JobScheduler-Service] Refactor models and synchronize models across services

### PR DESCRIPTION
**Pull request by Lucas-Angelo**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

# Type of change
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance improvement (change that optimizes software performance)
- [ ] Code refactoring (restructuring of the source code without changing external functionality)
- [ ] Usability enhancement (change that improves the user experience)
- [ ] Dependency update (updating libraries or external components)
- [ ] Security fix (change that addresses a security vulnerability)
- [ ] Documentation (addition or update of software documentation)
- [ ] Removal of deprecated features (elimination of unused functionalities or components)
- [ ] Testing (addition or update of automated tests)
- [ ] Others (specify the type of change)

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Browser: (e.g., Google Chrome, Mozilla Firefox, Safari)
* Device: (e.g., desktop, laptop, tablet, smartphone)
* Screen Resolution: 
* Package Manager: (e.g., npm, Yarn)
* Docker Containerization: (e.g., Yes/No)

# Checklist:

## Backend Testing Checklist:
- [ ] Test endpoints using Postman:
  - [ ] Verify if endpoints are responding correctly.
  - [ ] Test different HTTP methods (GET, POST, PUT, DELETE, etc.).
  - [ ] Check returned status codes (200, 400, 500, etc.).
  - [ ] Send requests with different parameter sets.
  - [ ] Test authentication and authorization cases if applicable.
  - [ ] Verify the validity and consistency of returned data.
  - [ ] Create a Postman collection with all the necessary requests.
- [ ] Verify if the application functions correctly with Docker Compose.
- [ ] Verify if the application functions correctly without Docker Compose in the local environment.

## Frontend Testing Checklist:
- [ ] Responsiveness:
  - [ ] Test on different devices (desktop, laptop, tablet, smartphone).
  - [ ] Verify appearance and layout on different screen sizes.
  - [ ] Test functionality in both landscape and portrait mode on mobile devices.
- [ ] Browsers:
  - [ ] Test on popular browsers (Google Chrome, Mozilla Firefox, Safari).
  - [ ] Check compatibility and consistent behavior across different browsers.

## Automated Testing:
- [ ] Test Coverage: Evaluate the coverage of automated tests.
- [ ] Unit Tests: Execute unit tests to verify specific functionality.
- [ ] Integration Tests: Perform integration tests to check component interactions.

### CODE REVIEW

**Comment by Lucas-Angelo** [comment]
@guizombas open to code review, im creating tests in the meantime. 

obs: need merge #159 before.

File: Codigo/dtos-lib/dto/evaluationMethod.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  /**
+   * @isInt year must be an integer
+   */
+  year: number;
+  /**
+   * @isOptional disabledAt is optional
+   * @isDate disabledAt must be a Date
+   * @default null
+   */
+  disabledAt?: Date | null;
```
Comment (guizombas) [inline Codigo/dtos-lib/dto/evaluationMethod.ts]: faz sentido ter esse disabledAt na criação?

File: Codigo/supportplatform-service/src/controller/RepositoryController.ts [hunk 2]
```diff
@@ -2,25 +2,41 @@ import {
...
+    Tags,
+} from "tsoa";
 import { RepositoryMapper } from "../mapper/RepositoryMapper";
 import RepositoryService from "../service/RepositoryService";
 
 @Route("repository")
 @Security("bearer", ["admin"])
 @Tags("repository")
-export class RepositoryController {
+export class RepositoryController extends Controller {
```
Comment (guizombas) [inline Codigo/supportplatform-service/src/controller/RepositoryController.ts:26]: pergunta sincera, o que impacta extender da classe Controller (e chamar o super no construtor igual fez)?

**Review by guizombas — APPROVED** [review]
Deixei só dois pequenos comentários, mas está aprovado mesmo com eles.

File: Codigo/supportplatform-service/src/controller/RepositoryController.ts (hunk 2 above)
Comment (guizombas) [reply Codigo/supportplatform-service/src/controller/RepositoryController.ts:26]: vi que tem a ver com o this.setStatus(200), o que ele impacta?

File: Codigo/dtos-lib/dto/evaluationMethod.ts (hunk 1 above)
Comment (Lucas-Angelo) [reply Codigo/dtos-lib/dto/evaluationMethod.ts]: removerei

File: Codigo/supportplatform-service/src/controller/RepositoryController.ts (hunk 2 above)
Comment (Lucas-Angelo) [reply Codigo/supportplatform-service/src/controller/RepositoryController.ts:26]: @guizombas possibilita definir algumas coisas da requisição, como `this.setStatus(201);`

File: Codigo/dtos-lib/dto/evaluationMethod.ts (hunk 1 above)
Comment (guizombas) [reply Codigo/dtos-lib/dto/evaluationMethod.ts]: pode dar resolve aqui nesse comentário depois que tiver subido

File: Codigo/supportplatform-service/src/controller/RepositoryController.ts (hunk 2 above)
Comment (guizombas) [reply Codigo/supportplatform-service/src/controller/RepositoryController.ts:26]: tendi tendido

**Pull request merged**